File: frontend/src/types/queue.ts
import type { Alert } from './analysis';

export interface PriorityQueueItem {
  rank: number | null;
  alert_id: string;
  water_body_id: string;
  zone_id: string;
  scene_id: string | null;
  analysis_date: string;
  investigation_priority_score: number;
  priority_band: string;
  severity: string;
  confidence: number;
  primary_driver: string | null;
  alert_details: Alert;
  investigation_status: string;
  updated_at: string;
  created_at: string;
}

export interface AlertStatusHistory {
  id: string;
  alert_id: string;
  previous_status: string;
  new_status: string;
  changed_at: string;
  changed_by: string;
  reason: string | null;
}

export interface QueueResponse {
  items: PriorityQueueItem[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface QueueSummary {
  active: number;
  acknowledged: number;
  under_investigation: number;
  resolved: number;
  very_high_priority: number;
  high_priority: number;
}

File: backend/app/schemas/queue.py
from pydantic import BaseModel
from typing import List, Optional, Any
from app.schemas.alert import AlertSchema

class PriorityQueueItem(BaseModel):
    rank: Optional[int] = None
    alert_id: str
    water_body_id: str
    zone_id: str
    scene_id: Optional[str] = None
    analysis_date: str

    investigation_priority_score: float
    priority_band: str

    severity: str
    confidence: float

    primary_driver: Optional[str] = None
    
    # We will reuse the full alert object as the underlying data, 
    # but the frontend might just use the QueueItem. Let's include
    # the original AlertSchema for full detail view context without
    # duplicating everything.
    alert_details: AlertSchema

    investigation_status: str
    updated_at: str
    created_at: str

class AlertStatusUpdate(BaseModel):
    status: str
    reason: Optional[str] = None

class AlertStatusHistorySchema(BaseModel):
    id: str
    alert_id: str
    previous_status: str
    new_status: str
    changed_at: str
    changed_by: str
    reason: Optional[str] = None

class QueueResponseSchema(BaseModel):
    items: List[PriorityQueueItem]
    total: int
    page: int
    page_size: int
    total_pages: int

class QueueSummarySchema(BaseModel):
    active: int
    acknowledged: int
    under_investigation: int
    resolved: int
    very_high_priority: int
    high_priority: int
